panel/io/jupyter_server_extension.py

```diff
@@ -389,6 +389,9 @@ async def open(self, path, *args, **kwargs) -> None:
         self.session_id = get_session_id(token)
         if self.session_id not in state._kernels:
             self.close()
+            msg = f"Session ID '{self.session_id}' does not correspond to any active kernel."
+            raise ProtocolError(msg)
```

Comment:
Yes, indeed, this should be logged. I would raise a simple `RuntimeError` here though, it's not ProtocolError since all the protocol validation has been executed correctly by this point.